### PR TITLE
Add a helper to use Inputs as messages

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | na
| License       | MIT
| Doc PR        | wiip

### CODE REVIEW

File: src/Bridge/Symfony/Messenger/DataTransformer/InputMessageDataTransformer.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function transform($object, string $to, array $context = [])
+    {
+        return $object;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsTransformation($object, string $to, array $context = []): bool
```
Comment (GregoireHebert) [inline src/Bridge/Symfony/Messenger/DataTransformer/InputMessageDataTransformer.php]: Not really related to this but I suggest changing `$object` with `$data` while it's still possible. It's a bit confusing.

**Comment by dunglas** [comment]
I like the idea, but I'm not a fond of the interface. Couldn't we use a new value for the `messenger` attribute instead? `messeger="input"` for instance?

**Comment by soyuka** [comment]
Done @dunglas wdyt?

File: src/Bridge/Symfony/Messenger/DataTransformer.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            return false;
+        }
+
+        $metadata = $this->resourceMetadataFactory->create($context['resource_class'] ?? $to);
+
+        if (!isset($context['operation_type'])) {
+            return 'input' === $metadata->getAttribute('messenger');
+        }
+
+        return 'input' === $this->resourceMetadataFactory->create($context['resource_class'] ?? $to)
```
Comment (dunglas) [inline src/Bridge/Symfony/Messenger/DataTransformer.php]:
```suggestion
        return 'input' === $metadata
```

**Comment by dunglas** [comment]
looks good to me

**Pull request merged**